src/telemetry.rs

```diff
@@ -12,11 +12,19 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use crate::proxy::Error;
```

Comment:
we shouldn't use proxy::Error; we can define our own if needed here.

Reply:
use a new error instead proxy:Error